ext/intl/formatter/formatter_format.c

```diff
@@ -135,11 +184,20 @@ PHP_FUNCTION( numfmt_format_currency )
 	int32_t    scurrency_len = 0;
 	FORMATTER_METHOD_INIT_VARS;
 
+	object = getThis();
+
 	/* Parse parameters. */
-	if( zend_parse_method_parameters( ZEND_NUM_ARGS(), getThis(), "Ods",
-		&object, NumberFormatter_ce_ptr,  &number, &currency, &currency_len ) == FAILURE )
-	{
-		RETURN_THROWS();
+	if (object) {
+		ZEND_PARSE_PARAMETERS_START(2, 2)
+			Z_PARAM_STR_OR_NUMBER(number)
+			Z_PARAM_STRING(currency, currency_len)
+		ZEND_PARSE_PARAMETERS_END();
+	} else {
+		ZEND_PARSE_PARAMETERS_START(3, 3)
+			Z_PARAM_OBJECT_OF_CLASS(object, NumberFormatter_ce_ptr)
+			Z_PARAM_STR_OR_NUMBER(number)
+			Z_PARAM_STRING(currency, currency_len)
+		ZEND_PARSE_PARAMETERS_END();
 	}
```

Comment:
Why does the currency method/function support strings here? That seems to make no sense.

Reply:
It’s already in a separate commit, so I could remove it again, but I don’t see why it doesn’t make sense? Currency amounts can exceed int/float precision just like other numbers can…

Reply:
This function is to deal with the currency formatter, so using the decimal formatter somewhere under the hood is not what I expect.

Also I highly doubt one can have a currency amount of 18 quintillion (/trillion) overflow, even if that is in cents.

Reply:
Alright, currency part removed.